src/services/profile/profile-router.ts

```diff
@@ -72,4 +76,235 @@ profileRouter.get("/leaderboard/", async (req: Request, res: Response) => {
+ * @apiGroup Profile
+ * @apiDescription Retrieve the user's profile based on the provided ID as a path parameter.
+ *
+ * @apiParam {String} id User's unique ID.
+ *
+ * @apiSuccess (200: Success) {Json} user User's profile information.
+ * @apiSuccessExample Example Success Response:
+ * HTTP/1.1 200 OK
+ * {
+ *    "_id": "12345",
+ *    "firstName": "Hackk",
+ *    "lastName": "Illinois",
+ *    "discord": "hackillinois",
+ *    "avatarUrl": "na",
+ *    "points": 0,
+ *    "id": "abcde",
+ *    "foodWave": 0
+ * }
+ *
+ * @apiError (404: Not Found) {String} UserNotFound The user's profile was not found.
+ * @apiError (500: Internal Error) {String} InternalError An internal server error occurred.
+ * @apiErrorExample Example Error Response (UserNotFound):
+ *     HTTP/1.1 404 Not Found
+ *     {"error": "UserNotFound"}
+ *
+ * @apiErrorExample Example Error Response (InternalError):
+ *     HTTP/1.1 500 Internal Server Error
+ *     {"error": "InternalError"}
+ */
+
+profileRouter.get("/id/:id", weakJwtVerification, async (req: Request, res: Response) => {
+    // const collection: Collection = databaseClient.db(Constants.PROFILE_DB).collection(ProfileDB.PROFILES);
+
+    const id: string | undefined = req.params.id;
+
+    const user: AttendeeProfile | null = await AttendeeProfileModel.findOne({ userId: id });
+
+    if (!user) {
+        return res.status(Constants.NOT_FOUND).send({ error: "UserNotFound" });
+    }
+
+    return res.status(Constants.SUCCESS).send(user);
+});
+
+/**
+ * @api {post} /profile POST /profile
+ * @apiGroup Profile
+ * @apiDescription Create a user profile based on their authentication.
+ *
+ * @apiBody {String} firstName User's first name.
+ * @apiBody {String} lastName User's last name.
+ * @apiBody {String} discord User's Discord username.
+ * @apiBody {String} avatarUrl User's avatar URL.
+ *
+ * @apiSuccess (200: Success) {Json} user Created user's profile information.
+ * @apiSuccessExample Example Success Response:
+ * HTTP/1.1 200 OK
+ * {
+ *    "_id": "abc12345",
+ *    "firstName": "Hack",
+ *    "lastName": "Illinois",
+ *    "discord": "HackIllinois",
+ *    "avatarUrl": "na",
+ *    "points": 0,
+ *    "id": "12345",
+ *    "foodWave": 0
+ * }
+ *
+ * @apiError (400: Bad Request) {String} UserAlreadyExists The user profile already exists.
+ * @apiError (500: Internal Error) {String} InternalError An internal server error occurred.
+ * @apiErrorExample Example Error Response (UserAlreadyExists):
+ *     HTTP/1.1 400 Bad Request
+ *     {"error": "UserAlreadyExists"}
+ *
+ * @apiErrorExample Example Error Response (InternalError):
+ *     HTTP/1.1 500 Internal Server Error
+ *     {"error": "InternalError"}
+ */
+
+profileRouter.post("/", strongJwtVerification, async (req: Request, res: Response) => {
+    const profile: AttendeeProfile | null = req.body as AttendeeProfile;
+    if (!profile) {
+        return res.status(Constants.BAD_REQUEST).send({ error: "InvalidPostData" });
+    }
+
```

Comment:
I'd create a new format in profile-formats for this, and add a function to verify that the returned params are actually right